src/core/qgsvectordataprovider.h

```diff
@@ -554,8 +554,9 @@ class CORE_EXPORT QgsVectorDataProvider : public QgsDataProvider, public QgsFeat
      * In particular on the OGR provider, a pooled connection will be invalidated.
      * This forces QGIS to reopen a file or connection.
      * This can be required if the underlying file is replaced.
-     */
-    virtual void forceReload();
+     * \deprecated QGIS 3.12 - will be removed in QGIS 4.0 - use reloadData instead
```

Comment:
I would suggest to remove all the rest of the docs above to have one canonical source for information on reloadData